### ISSUE TITLE
chown messing up read-only volume mounts

### ISSUE THREAD

**Issue by kwmonroe** [juju-solutions/charmbox#60]
I'm on OSX 10.11.6 using Docker for Mac 1.12.1 (build: 12133).

If i volume mount a read-only file into charmbox, docker responds with Permission denied:

```
$ touch /tmp/foo
$ chmod 400 /tmp/foo
$ ls -la /tmp/foo
-r--------  1 kwmonroe  wheel  0 Oct  3 17:36 /tmp/foo
$ docker run -ti --rm -v /tmp/foo:/home/ubuntu/foo jujusolutions/charmbox
chown: changing ownership of '/home/ubuntu/foo': Permission denied
$ echo $?
1
```

Couple interesting things here.  First, I can mount that read-only file anywhere in the docker container except `/home/ubuntu`:

```
$ docker run -ti --rm -v /tmp/foo:/home/foo jujusolutions/charmbox
welcome to juju
ubuntu@ca3293fe052d:~$ ls -la /home/foo
-r-------- 1 ubuntu ubuntu 0 Oct  3 22:36 /home/foo
```

Second, if I comment out the chown here:

https://github.com/juju-solutions/charmbox/blob/devel/run.sh#L11

And rebuild a local charmbox, I can mount to /home/ubuntu once again:

```
$ docker run -ti --rm -v /tmp/foo:/home/ubuntu/foo charmbox
welcome to juju
ubuntu@75dea0bcd3af:~$ ls -la /home/ubuntu/foo 
-r-------- 1 ubuntu ubuntu 0 Oct  3 22:36 /home/ubuntu/foo
```

Fwiw, this happens on `latest` and `devel` versions of charmbox.  Soooo, is that `chown` necessary?  Any idea why chowning `/home/ubuntu` would affect my ability to volume mount a read-only file there?

This is a real bummer when volume mounting source directories that contain git repositories, as lots of .git files are read-only (eg: `.git/objects/pack/*`).

**Comment by lazypower** [comment]
@kwmonroe I don't think there are any major impacts to removing that line, did charmbox continue to function as intended?

**Comment by mbruzek** [comment]
@simonklb added the -R recently and I merged it. I believe he did so because he was using charmbox without volume mounts and the directories were owned by root https://github.com/juju-solutions/charmbox/issues/57. Please look at https://github.com/juju-solutions/charmbox/pull/58 for the changes.

Can we get to the root cause and find a solution that works fro everyone?

**Comment by simonklb** [comment]
This is a big problem using volumes with Docker overall. The fact that the UID used in the Docker container impacts the permissions of the volumes on the host can mess up things.

Since youre more or less bound to run the charmbox container as UID 1000 perhaps it's best to manually chown things instead of letting the script do it. Or check if the volumes are empty and only do it in that case.

Feel free to revert the commit if it's causing a lot of trouble.

**Comment by chuckbutler** [comment]
I think we all need a read of https://denibertovic.com/posts/handling-permissions-with-docker-volumes/

**Comment by lazypower** [comment]
Thanks for the assist on this one everyone! It appears this was closed in #61
